jsFiles/utility.js

```diff
@@ -26,35 +26,35 @@ function search(event) {
   //prevent the default action of the form (reloading the page)
   const cityRequested = inputElement.value;
   const api = `${apiRootUrl}weather?q=${cityRequested}&appid=${KEY}&units=metric`;
-  fetchData(api);
+  // true parameter displays the time of the searched city
+  fetchData(api, true);
 }
 
 /**
  * displays date and time
  *
  * @param {object} dateObj - contains date and time information
  */
-// TODO: fix for when displaying the time of a searched city -- timezones offset
 function dateTimeDisplay(dateObj) {
   const date = `${dateObj.getDate()}/${
     dateObj.getMonth() + 1
   }/${dateObj.getFullYear()}`;
 
-  // en-US: uses 12-hour time with AM/PM
-  const time = dateObj.toLocaleTimeString("en-US", {
-    hour: "2-digit",
-    minute: "2-digit",
-  });
-  dateTimeElement.innerHTML = `Today's date: ${date}, time: ${time}`;
+  const time = formatTime(dateObj);
+  localDateTimeElement.innerHTML = `Local date: ${date}, local time: ${time}`;
 }
 
 /**
- * handles the alert error messages of fetchData() and geolocation.getCurrentPosition()
+ * this function calculates the time of the searched city
+ *  and returns a displayable time
  *
- * @param {object} error - contains information about the error
+ * @param {object} dateObj - contains date and time information
+ * @param {number} offsetTime - units of seconds
  */
-function alertError(error) {
-  alert(error.message);
+function cityTime(dateObj, offsetTime) {
+  // .getTime() returns miliseconds
+  const dateObj1 = new Date(dateObj.getTime() + offsetTime * 1000);
```

Comment:
1. don't name things something1, something2. It's not a good code style and don't bring value :) 
Better name could be convertedTime, dateWithOffset and so on.
2. you don't need this extra variable you can do all in one line:
```js
  return `Time: ${formatTime((new Date(dateObj.getTime() + offsetTime * 1000)), "UTC")}`;
```